client/apps/edit2/components/content/section_controls/index.jsx

```diff
@@ -34,143 +44,122 @@ export class SectionControls extends Component {
+      const scrollPosition = window.scrollY
 
-  sectionIsImage () {
-    return this.props.section.get('type').includes('image')
-  }
+      const scrollPlusControls = scrollPosition + controlsHeight
+      const scrollPlusSection = offsetTop + sectionHeight
 
-  sectionHasFullscreen (section) {
-    const hasFullscreen = ['embed', 'video'].includes(section.get('type')) || this.sectionIsImage()
-    return this.props.articleLayout === 'feature' && hasFullscreen
+      return scrollPlusControls > scrollPlusSection
+    }
   }
 
-  renderSectionLayouts (section) {
-    return (
-      <nav className='edit-controls__layout'>
-        <a
-          name='overflow_fillwidth'
-          className='layout'
-          onClick={() => this.changeLayout('overflow_fillwidth')}
-          data-active={section.get('layout') === 'overflow_fillwidth'} />
-        <a
-          name='column_width'
-          className='layout'
-          onClick={() => this.changeLayout('column_width')}
-          data-active={section.get('layout') === 'column_width'} />
-        {
-          this.sectionHasFullscreen(section) &&
-          <a
-            name='fillwidth'
-            className='layout'
-            onClick={() => this.changeLayout('fillwidth')}
-            data-active={section.get('layout') === 'fillwidth'}>
-            <IconImageFullscreen fill={'white'} />
-          </a>
-        }
-        {
-          this.hasImageSet() &&
-          <a
-            name='image_set'
-            className='layout'
-            onClick={this.toggleImageSet}
-            data-active={section.get('type') === 'image_set'} />
+  insideComponent = () => {
+    const { isHero } = this.props
+    if (this.controls) {
+      const $section = $(this.controls).closest('section')
+
+      let insideComponent = false
+      const isScrollingOver = this.isScrollingOver($section)
+      const isScrolledPast = this.isScrolledPast($section)
+
+      if ($section && !isHero) {
+        if (
+          (isScrollingOver && !isScrolledPast) ||
+          (isHero && !isScrollingOver)
+        ) {
+          insideComponent = true
         }
-      </nav>
-    )
+      }
+      return insideComponent
+    }
   }
 
   render () {
-    const { isHero, section, sectionLayouts } = this.props
+    const {
+      children,
+      disabledAlert,
+      isHero,
+      showLayouts
+    } = this.props
     const { insideComponent } = this.state
-    const isSticky = insideComponent ? ' sticky' : ''
+
     const outsidePosition = isHero ? 'relative' : 'absolute'
+    const position = insideComponent ? 'fixed' : outsidePosition
+    const bottom = this.getPositionBottom()
 
     return (
-      <div
-        ref='controls'
-        className={'edit-controls' + isSticky}
-        style={{
-          position: insideComponent ? 'fixed' : outsidePosition,
-          bottom: this.getPositionBottom()
+      <SectionControlsContainer
+        innerRef={(node) => { this.controls = node }}
```

Comment:
What does `innerRef` refer to?

Reply:
innerRef is required to use refs with StyledComponents